### PR TITLE
Simplification hints api needs a little more flexibility

### PR DESCRIPTION
**Pull request by longrunningprocess**
`Flourish` covers both scenarios, case-insensitivity and missing sense.

image

`LIKE` is protected

image

wildcard is not the default

image

but `%` is supported

image

### CODE REVIEW

**Review by craigp-atw — CHANGES_REQUESTED** [review]
Using `LIKE ${term}%` results in a lot of false positives based on the first letters of words. See the search for `li` below. I'd rather we be more intentional about slitting the sense and look for equality of the entire stem.
image

**Comment by longrunningprocess** [comment]
> Using `LIKE ${term}%` results in a lot of false positives based on the first letters of words. See the search for `li` below. I'd rather we be more intentional about splitting the sense and look for equality of the entire stem. image

good call, I'll narrow it

**Comment by longrunningprocess** [comment]
take another look that narrows but also permits wildcards if requested.

File: app/src/lib/server/ontology.js [hunk 1]
```diff
@@ -90,22 +90,32 @@ export async function get_version(db) {
...
+ * @returns {(term: string) => Promise<SimplificationHint[]>} term is case-insensitive
  */
 export const get_simplification_hints = db => async term => {
+	if (!term) {
+		return []
+	}
+
+	const ENDS_IN_A_SENSE = /-[A-Z]$/
+	if (!ENDS_IN_A_SENSE.test(term)) {
+		term = `${term}-A`
```
Comment (craigp-atw) [inline app/src/lib/server/ontology.js:102]: This isn't right either. Now it doesn't work for words like 'accurate' that don't have a sense within the how-to document. I think we might need to change the import from the how-to document to split the stem and sense and store them separately in the db like we do in the ontology.

**Review by craigp-atw — CHANGES_REQUESTED** [review]

File: app/src/lib/server/ontology.js (hunk 1 above)
Comment (longrunningprocess) [reply app/src/lib/server/ontology.js]: fix here:  [6d53715](https://github.com/presciencelabs/tabitha-ontology/pull/25/commits/6d53715e1777d83afcd76ae5e73b5166e9c32cba)

File: app/src/lib/server/ontology.js [hunk 2]
```diff
@@ -93,26 +93,50 @@ export async function get_version(db) {
...
+
+	const term_w_default_sense = `${term}-A`
+	/**
+	 * handles the following cases:
+	 *		- complex_term=flourish
+	 *		- complex_term=accuse (results in multiple rows for accuse-A but not accuse-B)
+	 *
+	 * @type {import('@cloudflare/workers-types').D1Result<SimplificationHint>}
+	 */
+	const { results: results_second_try } = await db.prepare(sql).bind(term_w_default_sense).all()
```
Comment (craigp-atw) [inline app/src/lib/server/ontology.js:137]: What about `${term}-_` for the like query? I still don't like that we default to A. `enable` is another case where there are A/B senses

**Review by craigp-atw — APPROVED** [review]

**Pull request merged**